kernel/timeout.c

```diff
@@ -244,17 +244,19 @@ void sys_clock_announce(int32_t ticks)
 
 	k_spinlock_key_t key = k_spin_lock(&timeout_lock);
 
+#if defined(CONFIG_SMP)
```

Comment:
This bit looks like a noop?  announce_remaining should be guaranteed positive-definite, so the swap of > vs. != doesn't matter.  If there's concern there might be a wraparound bug, an __ASSERT() would be better.  Also just style: IS_ENABLED() reads more cleanly to my eyes in situations like this where there isn't an undefined symbol or whatnot.

Reply:
I'll make the switch back the original forms. That should alleviate future questions such as "why exactly did we switch to '>' from '!=' ?".  The CONFIG_SMP change is the byproduct of a bit of experimentation I reverted where I was evaluating two minor performance tweaks to this early bailout (but they are in fact out of scope of this fix).